### DISCUSSION TITLE
New account tree => permissions nightmare. Reinstall?

### DISCUSSION THREAD

**Discussion by JosN1** [Homebrew/discussions#1960] [Everyday usage] [answered]
Semi-old Macbook Pro. There used to be two accounts on this laptop, an administrative account and a standard (limited) user account. They were both mine. When I was in the user account, I would SU - [admin name] to install brew, work on brew, etc. As I became more familiar with *nix systems, I made my user account an admin account and deleted the previous admin account. When I first went to sudo brew, the cautionary messages said not to do that, so I now don't. I chowned the directories that brew told me to chown. It mostly works now. But not entirely....  There are multiple directories that I have insufficient permissions for brew to upgrade most (but not all) apps. It would take too long for me to chown them all. But do I have a choice? Can I uninstall brew, although the specific account ID that installed it is no longer available? What's the best approach here?

**Comment by carlocab** [comment, marked as answer]
Have you tried doing
```
sudo chown -R $(whoami) $(brew --prefix)/*
```
?

**Reply by JosN1** [reply]
No. I don't really understand that syntax, what the parameters after (whoami), do. But... I'll try it anyway and report back :-) Thank you.

**Reply by alebcay** [reply]
For reference we're changing the ownership of the things in the Homebrew prefix:
- `sudo`: Elevate and run as root (needed since the things in the Homebrew prefix may be owned by other users).
- `chown`: **ch**ange **own**er.
- `-R`: recursively.
- `$(whoami)`: who to set as the new owner. We run the command `whoami` which should return your user; the `$()` syntax captures the output and substitutes it here.
- `$(brew --prefix)/*`: glob representing files to change ownership of. `brew --prefix` is a command that should return the Homebrew prefix; again, we're capturing that output with `$()`. Then we want to select all the things in that folder, hence the `/*` suffix.

**Reply by JosN1** [reply]
Thank you for the explanation. That last statement was new to me. It's useful. I've marked your response as an answer. It worked perfectly.
